src/Content.php

```diff
@@ -0,0 +1,167 @@
+<?php
+
+namespace Graby;
+
+/**
+ * Immutable container for Graby result.
+ */
+final class Content
+{
+    private int $status;
+
+    private string $html;
+
+    private string $title;
+
+    private ?string $language;
+
+    private ?string $date;
+
+    /** @var string[] */
+    private array $authors;
+
+    private string $url;
+
+    private ?string $image;
+
+    /** @var array<string, string> */
+    private array $headers;
+
+    private bool $isNativeAd;
+
+    private ?string $summary;
+
+    /**
+     * @param string[]              $authors
+     * @param array<string, string> $headers
+     */
+    public function __construct(
+        int $status,
+        string $html,
+        string $title,
+        ?string $language,
+        ?string $date,
+        array $authors,
+        string $url,
+        ?string $image,
+        array $headers,
+        bool $isNativeAd,
+        ?string $summary = null
+    ) {
+        $this->status = $status;
+        $this->html = $html;
+        $this->title = $title;
+        $this->language = $language;
+        $this->date = $date;
+        $this->authors = $authors;
+        $this->url = $url;
+        $this->image = $image;
+        $this->headers = $headers;
+        $this->isNativeAd = $isNativeAd;
+        $this->summary = $summary;
+    }
+
+    public function getStatus(): int
+    {
+        return $this->status;
+    }
+
+    public function getHtml(): string
+    {
+        return $this->html;
+    }
+
+    public function withHtml(?string $html): self
+    {
+        $new = clone $this;
```

Comment:
I'm wondering if this won't use too much ram because of intense cloning during the process (with all these `with*` methods used)?

Reply:
I do not think it would, it is just a shallow copy. And during the chaining, the original copies should be GC'd almost immediately, if the JIT compiler does not eliminate the cloning altogether.